internal/handlers/pg/msg_update.go

```diff
@@ -100,6 +100,12 @@ func (h *Handler) MsgUpdate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		if u, err = common.GetOptionalParam(update, "u", u); err != nil {
 			return nil, err
 		}
+		if u != nil {
```

Comment:
Before this PR, here the check of the upsert document was after the data was loaded to the client.
So if the data volume is big,  and `u` condition document is wrong, then the client unnecessarily waited for data.